### PR TITLE
refactoring to not use context + fixup tests

### PR DESCRIPTION
**Pull request by ssunday**
some issues with the tests/flow...it has knowledge that it really shouldn't—like board knows it has 9 squares—why? the test is really that it renders the length of squares provided to it. My types were a bit sloppy

### CODE REVIEW

File: src/Game.tsx [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
-type Props = {};
+type State = {
+  board: string[],
+  xToPlay: boolean
+};
 
-export const Game = (props: Props): JSX.Element => {
-  const state = useGameState();
+export class Game extends React.Component<{}, State> {
+  public constructor(props: {}) {
```
Comment (beporter) [inline src/Game.tsx:10]: When I was reading the React docs, their introduction of Hooks makes it sound like that FunctionalComponents are the future and classes are a "legacy" approach. That's why I was trying to stay away from them. https://reactjs.org/docs/hooks-intro.html#no-breaking-changes

Turning the chair around though; I can see how it'd be useful to be familiar with both ways of doing when joining existing projects/codebases. This ecosystem moves so fast that FC's will probably be out of style by next year anyway in favor of something else. 😇

**Comment by beporter** [comment]
This PR is still `wip` correct? I don't want to comment on anything you're still working on.

**Comment by ssunday** [comment]
@beporter I mean, it's a proof of concept, I don't intend to do anymore with it.

File: src/Game.tsx (hunk 1 above)
Comment (ssunday) [reply src/Game.tsx:10]: @beporter Yeah, we work with legacy codebases, so hooks aren't in use with them. It's good to learn about context/hooks, but their applicability in our code-bases is limited. That's why I was pushing you _not_ to. And they're new in 16.8, which I don't think we're using for most of our code bases....

And tbh, with react, I'd like to wait a few version before using those bits in prod.

(also, full disclosure, I had no idea about hooks into you brought it up)

**Comment by beporter** [comment]
>I mean, it's a proof of concept, I don't intend to do anymore with it.

I spotted a regression (you can play "over" other pieces in this branch) and I assume files like `GameContext` aren't actually active anymore and can be removed (just wasn't important in order to demonstrate the refactor). But I have no problem just using this branch and fixing those issues up in order to proceed forward if that's okay with you.

**Comment by ssunday** [comment]
@beporter Yeah, I didn't remove the old stuff, I was just demonstrating how to use state instead of context in this case. I didn't intend to rebuild the entire app for you haha

**Pull request closed without merging**